ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -555,6 +558,20 @@ let definition_query (state : State.t) uri position merlin_request =
   let+ result = Document.dispatch_exn doc command in
   location_of_merlin_loc uri result
 
+let workspace_symbol (state : State.t) (params : WorkspaceSymbolParams.t) =
+  let rootUri = (init_params state).rootUri in
+  let symbols = Workspace_symbol.run params rootUri in
+  match symbols with
+  | Ok symbols -> Fiber.return (Some symbols)
+  | Error e -> (
+    match e with
+    | Workspace_symbol.No_build_dir path ->
+      Jsonrpc.Response.Error.raise
```

Comment:
Is it the correct place to raise an error? Also, `InternalError` is probably not the good error code 🤔

Reply:
Actually, I'm not even sure we should raise an error in this case. I guess it will happen mostly because the user did not run the build command yet. Maybe we should display a notification instead to inform the user?

Reply:
Internal error is definitely not right. A message asking the user to build the project first seems fine.